enhancements/running-on-the-host.md

```diff
@@ -0,0 +1,167 @@
+# Running on the host
+
+## Summary
+
+This enhancement proposes an operating mode of OpenPERouter where the router is able to start at boot time and to provide connectivity to the VPN network before the Kubernetes machinery starts.
+
+## Motivation
+
+Running the nodes completely out of VPN does not work with the current design. In order to configure the router, we need to fetch the configuration from the API server, but in order to reach the API server we need to configure the router.
+
+To solve this chicken-egg problem we need an operating mode where the router is allowed to start as soon as possible and to be fed with a static configuration used to provide the basic connectivity required to bootstrap kubernetes.
+
+## Goals
+
+- Running the router before the kubelet and the cluster's primary CNI
+- Providing a way to configure the router statically
+- Providing a way to extend the static configuration with a configuration provided via the current kubernetes API
+
+## Non Goals
+
+- Having an alternative way to provide the static configuration that involves distritbuting the configuration to the routers
+
+## Proposal
+
+### User stories
+
+- As a cluster admin, I want to deploy the router to the nodes of the cluster with enough configuration to allow it to provide the basic connectivity.
+
+- As a cluster admin, I want to use Kubernetes CRDs to provision additional networks to reach / expose different classes of services or implement multitenancy.
+
+### Design Details
+
+The general idea is to run the same set of containers we have in pods today, but on the host. This includes:
+
+- running the router pod, in charge of providing the network control plane with FRR and the data plane with veths and the underlay interface
+- running the controller pod, in charge of reading the static configuration and receiving the additional configuration when the kubernetes api is available
+
+#### Handling the lifecycle of the components
+
+The router and the controller need to run as systemd units. This allows us to run them before the kubelet and the CNI start.
```

Comment:
Have you check if static pods fits this so we don't need external containers ? as far as I know if static pods use hostNetwork: true they run before CNI which is exactly what we need, we will have the pods running as part of kubernetes but we break the chicken egg problem that we have.

Reply:
This might be already late in the lifecycle, depending on the assumptions made by the CNI on how to deploy it. For example, OVN-Kubernetes relies on the default gateway to be setup, way earlier than the kubelet starts